src/Native/System.Native/pal_io.cpp

```diff
@@ -359,6 +359,39 @@ int32_t FcntlSetPipeSz(int32_t fd, int32_t size)
 }
```

Comment:
Shouldn't this native code be split out between OSs? This seems applicable to only *NIX/OSX.